src/Model/Serializer/ValueObject/Calendar/CalendarSerializer.php

```diff
@@ -87,6 +87,15 @@ function ($openingHourData) {
             );
         }
 
+        // There are cases where the calendar type does not match the number of sub events.
+        // For those cases the amount of sub events dictates the calendar type.
+        if ($calendarType->sameAs(CalendarType::single()) && count($subEvents) > 1) {
```

Comment:
Wouldn't a better solution be creating new events in the event store to fix these events?

Reply:
This code is used to read the JSON payload from the existing and new events inside the event store. The new events which are created with commands with better validation will have the correct structure.